### PR TITLE
Feature/system vars check

### LINKED ISSUE #62: 'Missing environment variable: ' when variable is defined as system variable but not .env file
Obviously this assumes systemvars is enabled.

Looking through the code, 
'env' contains variables from .env
'vars' contains variables from process.env

The 'Missing environment variable' error is thrown if any value from the blueprint is not in 'env'.
I believe it should also check for the key in 'vars'

Sounds like a very simple fix. Do I understand this correctly?

### PR DESCRIPTION
**Pull request by mrsteele**
closes #62

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ class Dotenv {
     }
 
     Object.keys(blueprint).map(key => {
-      const value = env[key]
+      const value = env[key] || vars[key]
       if (!value && options.safe) {
```
Comment (JordanBelford) [inline src/index.js]: This conditional should also handle case where value is falsey.

File: src/index.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ class Dotenv {
     }
 
     Object.keys(blueprint).map(key => {
-      const value = env[key]
+      const value = env[key] || vars[key]
```
Comment (mrsteele) [inline src/index.js]: Can you verify that? I think dotenv variables are always strings and a string with a 0 doesn't return falsy.

Try this:
'''
console.log("0" || "we got a problem")
'''

Comment (JordanBelford) [inline src/index.js]: In that case the only falsey value to consider is "", empty string.

If a system variable is set, can the dotenv override it with an empty value?

Comment (mrsteele) [inline src/index.js]: The way it is set up now you cannot. Looking at dotenv-safe it looks like as long as the variable is set to anything it passes the safe check.

So back to your question... perhaps we should allow you to override a system car locally even if it is an empty string.

I'll look into that this afternoon.

**Comment by mrsteele** [comment]
@JordanBelford I merged this in as everything on my end was working. Let me know if this resolves this problem and/or if anything else wrong occurs.

Good catch and I appreciate the feedback from others!

**Comment by JordanBelford** [comment]
@mrsteele Thank you! This works for me. 

Small edge case where it fails, if variable is set to empty string the "Missing environment variable" error is thrown.

Maybe do this?
```if ((!value || value === '') && options.safe) {```

**Comment by mrsteele** [comment]
@JordanBelford on the contrary, it looks like `dotenv-safe` by default also flags that. See here:

https://www.npmjs.com/package/dotenv-safe#allowemptyvalues

Perhaps an alternative (and something that would need some major refactoring) would be to piggy-back on dotenv-safe to make sure we can configure it further from dotenv-webpack.

**Pull request merged**